src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -433,5 +449,52 @@ void WiFiManager::PostConnectivityStatusChange(ConnectivityChange changeType)
     PlatformMgr().PostEventOrDie(&networkEvent);
 }
 
+System::Clock::Milliseconds32 WiFiManager::GetNextRecoveryTime()
+{
+
+    if (mConnectionRecoveryTimeMs >= kConnectionRecoveryMaxIntervalMs)
+    {
+        // reset previous connection recovery time to max value
+        mConnectionRecoveryTimeMs = kConnectionRecoveryMaxIntervalMs;
+        // find the new random jitter value in range [-jitter, +jitter]
+        int32_t jitter            = chip::Crypto::GetRandU32() % (2 * jitter + 1) - jitter;
+        mConnectionRecoveryTimeMs = mConnectionRecoveryTimeMs + jitter;
```

Comment:
I would just replace` mConnectionRecoveryTimeMs` with `kConnectionRecoveryMaxIntervalMs`, and remove the line 458.